engine/execution/ingestion/engine.go

```diff
@@ -430,8 +430,8 @@ func (e *Engine) executeBlock(
 	}
 
 	snapshot := e.execState.NewStorageSnapshot(*executableBlock.StartState,
-		executableBlock.ID(),
-		executableBlock.Block.Header.Height,
+		executableBlock.Block.Header.ParentID,
```

Comment:
This was wrong, when executing a block, the storage snapshot should be it's parent block, instead of the block itself, which has not been executed.